content/collections/web_experiment/en/implementation.md

```diff
@@ -108,6 +132,7 @@ Implement the `IntegrationPlugin` interface and set the `experimentIntegration`
 
 * `getUser(): object`: Return the [experiment user](/docs/feature-experiment/data-model#users) object.
 * `track(): boolean`: Track the event through a 3rd party. Return `true` if the event was tracked. Returning `false` causes the event to be persisted and retried at an interval.
+* `setup(): Promise<void>`: (Optional) Set up the integration asynchronously. Returns a promise which should be resolved when the integration is ready to return user information from `getUser()`.
```

Comment:
@bgiori Are there instances when the promise doesn't resolve when the integration is ready? Looking to use a a stronger word than "should" here.

Reply:
You're right, it's a "must" situation.